cpp/src/lance/format/lookup_table.h

```diff
@@ -21,7 +36,12 @@ class LookupTable {
 
   void AddPageLength(int32_t column, int32_t chunk, int64_t length);
 
-  int64_t GetOffset(int32_t column_id, int32_t chunk_id) const;
+  /// Get the file offset of a chunk of a array.
+  ///
+  /// \param column_id the column / field ID.
+  /// \param chunk_id the chunk id.
+  /// \return file offset if available.
+  std::optional<int64_t> GetOffset(int32_t column_id, int32_t chunk_id) const;
```

Comment:
when should we use arrow Result vs std::optional ?

Reply:
Result can include a reason which is nice to bubble up error messages to python?

Reply:
Yea, so `Result` has the "error" meaning, but optional has the semantic that "this value does not exist, but it is expected"? For this function, not all columns have offsets, i.e., the "parent" node is just a node without actual data.